### ISSUE TITLE
[Feature Request] Add 'symmetric' padding_mode (repeats boundary sample)

### ISSUE THREAD

**Issue by mtoksvig** [pytorch/pytorch#189236]
### 🚀 The feature, motivation and pitch

PyTorch's current 'reflect' padding_mode uses half-sample reflection, i.e. it **does not** repeat the boundary sample:
```
input: ABCDEF
reflect: DCB|ABCDEF|EDC  (boundaries not repeated)
```

I'd like to add a 'symmetric' padding_mode that **does** repeat the boundary sample:
```
input: ABCDEF
symmetric: CBA|ABCDEF|FED  (boundaries repeated)
```

This padding_mode matches e.g.:
- numpy's np.pad(..., mode='symmetric')
- OpenGL's GL_MIRRORED_REPEAT

Motivation: Symmetric padding better models physical conditions in many applications where the sample represents a measurement taken over an area (as opposed to a mathematical point). The current 'reflect' mode is appropriate for some use cases, but 'symmetric' is the more common convention in graphics and image processing

Proposed API:
```
F.pad(input, pad, mode='symmetric')
nn.Conv2d(..., padding_mode='symmetric')
```

Scope: This would add symmetric_pad{1,2,3}d ops (forward + backward) for CPU and CUDA, paralleling the existing reflection_pad implementation with a one-line change to the index formula.

Are you willing to open a PR? Yes.

### Additional context

to illustrate the difference, consider the weight assigned to each position when using with a 9-wide gaussian centered at positions 0 thru 4:

'reflect': note that position 1 gets more weight even when **centered** on position 0
<img width="1273" height="316" alt="Image" src="https://github.com/user-attachments/assets/7757debe-6938-4c15-bcf2-daa7ca80f3ff" />

'symmetric': note that weight keeps scrunching up against position 0 as you would expect
<img width="1269" height="307" alt="Image" src="https://github.com/user-attachments/assets/fa6b6879-3877-4786-8466-70a588d4fc80" />

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki
